### DISCUSSION TITLE
Strongswan Charon PID permissions issue with ipsec start command

### DISCUSSION THREAD

**Discussion by tomkonaut** [strongswan/strongswan#930] [Q&A]
Dealing with an issue where _ipsec restart | start_ from Ubuntu 18 / StrongSwan 5.6.2 yields a charon.pid within /var/run that has the following permissions:

-rw-r-----  1 root    root       6 Feb 25 23:23 charon.pid

This is problematic, because we use VRRP and check the availability of the .pid. If it doesn't exist, it doesn't assume the VIP.

Regardless, the ipsec start/restart commands create a pid without this permission. If I use the _systemctl start strongswan_ or _service strongswan start_ the pid is created with the correct permissions:

-rw-r--r--  1 root    root       6 Feb 25 23:23 charon.pid

Anyway to debug to see why this is happening? Or can help steer me in the right direction?

strongswan dpkg

ii  strongswan                             5.6.2-1ubuntu2.7                                all          IPsec VPN solution metapackage

**Comment by tobiasbrunner** [comment]
The permissions of a file don't have anything to do with other processes ability to check if that file exists. Relevant for that is the permission of the directory the file is in. Or is there an attempt to actually open and read the file? In that case you could apparently just not use `ipsec start` to fix the issue, no? Or what's the problem with using `systemctl start strongswan`?

**Reply by tomkonaut** [reply]
No doubt. No issues with using the systemctl method. It does pose a slight risk to some of the underlying designs that strongswan is being used for and I wanted to more intimately understand what the nature of the issue was. Regardless, I appreciate your input.

**Reply by tobiasbrunner** [reply]
> It does pose a slight risk to some of the underlying designs that strongswan is being used for

The legacy systemd unit simply calls `ipsec start` (with `--nofork`). So what do you mean?
